contracts/libs/0.8.x/BytecodeStorage.sol

```diff
@@ -70,79 +70,93 @@ library BytecodeStorage {
             // 0xf3    |  0xf3               | RETURN       |                                                                //
             //---------------------------------------------------------------------------------------------------------------//
+            // (11 bytes)
             hex"60_0B_59_81_38_03_80_92_59_39_F3",
             //---------------------------------------------------------------------------------------------------------------//
             // Opcode  | Opcode + Arguments  | Description  | Stack View                                                     //
             //---------------------------------------------------------------------------------------------------------------//
             // (1a) conditional logic for determing purge-gate (only the bytecode contract deployer can `selfdestruct`)
             //---------------------------------------------------------------------------------------------------------------//
-            // 0x60    |  0x60_14            | PUSH1 20     | 20                                                             //
-            // 0x60    |  0x60_35            | PUSH1 53 (*) | contractOffset 20                                              //
-            // 0x60    |  0x60_0C            | PUSH1 12     | 12 contractOffset 20                                           //
-            // 0x39    |  0x39               | CODECOPY     |                                                                //
-            // 0x33    |  0x33               | CALLER       | msg.sender                                                     //
-            // 0x60    |  0x60_20            | PUSH1 32     | 32 msg.sender                                                  //
-            // 0x52    |  0x52               | MSTORE       |                                                                //
-            // 0x60    |  0x60_00            | PUSH1 0      | 0                                                              //
-            // 0x51    |  0x51               | MLOAD        | byteDeployerAddress                                            //
-            // 0x60    |  0x60_20            | PUSH1 32     | 32 byteDeployerAddress                                         //
-            // 0x51    |  0x51               | MLOAD        | msg.sender byteDeployerAddress                                 //
-            // 0x14    |  0x14               | EQ           | (msg.sender == byteDeployerAddress)                            //
-            hex"60_14_60_35_60_0C_39_33_60_20_52_60_00_51_60_20_51_14",
+            // 0x60    |  0x60_14            | PUSH1 20           | 20                                                       //
+            // 0x60    |  0x60_4E            | PUSH1 76 (*)       | contractOffset 20                                        //
+            // 0x60    |  0x60_0C            | PUSH1 12           | 12 contractOffset 20                                     //
```

Comment:
Should we do the same thing here? (i.e. zero-pad the address in bytecode instead of using an offset)

Reply:
I don't know that we can do that IIUC, given how `CODECOPY` works (we need to copy into memory) but I am realizing that there is a place here where we can further optimize the assembly logic here. 

Pushing another update shortly here and will re-request a review – would love another look with the additional tweaks.

Reply:
Added additional optimization to just reference the `CALLER` directly from the stack when making the initial `(msg.sender == byteDeployerAddress)` comparison rather than `MSTORE`ing this into memory and then `MLOAD`ing it back out, which shaves off 6 op-codes.